parsel/selector.py

```diff
@@ -200,12 +200,15 @@ def re_first(
             return el
         return default
 
-    def getall(self) -> List[str]:
+    def getall(self, *, strip: bool = False) -> List[str]:
         """
         Call the ``.get()`` method for each element is this list and return
         their results flattened, as a list of strings.
         """
-        return [x.get() for x in self]
+        data = [x.get() for x in self]
```

Comment:
I think you can just pass `strip` to `get()` instead of essentially reimplementing the new code from get() here?
(granted, the code is almost trivial, but still)

Reply:
x is a Selector, at that point neither default nor strip exist anymore
Only SelectorList recognizes those two fields

Reply:
About adding the `strip` argument to the other overloads, wouldn't that remove the purpose of the overload on those? 
 Wouldn't having an overloaded `default + strip` be just going back to the non-overloaded method signature?

Reply:
> x is a Selector, at that point neither default nor strip exist anymore

Aren’t you also adding `strip` to `Selector`? Can’t you use `data = [x.get(strip=True) for x in self]`?

Reply:
That ended up looking harder than it should when I tried.
Also as far as I could see, default also is not on the Selector, so for standardization, I believe adding it would also be required. And in that case I don't understand why it wasn't added up until now, it gives me the feeling it is not a trivial task given the current implementation of get on Selector.

Reply:
This snippet displays the whole gist.

```
body = '<ul><li id="1">1</li><li id="2"> 2 </li><li id="3">3</li></ul>'
sel = self.sscls(text=body)
this_works = sel.xpath("//ul/li[position()>1]")[0].get()  # Using Selector
this_also_works = sel.xpath("//ul/li[position()>1]").get(default="6")  # Using SelectorList
this_doesnt_work = sel.xpath("//ul/li[position()>1]")[0].get(default="6")  # Selector doesnt have default field
````

I think that following the current approach of `default` for `strip` could be an easy win, as the following issue we could go forward with would be adding both of those fields directly to the `Selector`, which I don't think is a trivial task unless I am missing something out.

Reply:
Let me give this a look tomorrow... I think it may be feasible and I just got confused, let's see.

Reply:
Yeah, you are changing Selector.get() to have `strip` and it already has `default` so I think there shouldn't be any problems with using it, and it indeed doesn't look too complicated to me :)

Also if you have issues with typing I can hep, I'm working on improving parsel typing right now.

> this_doesnt_work = sel.xpath("//ul/li[position()>1]")[0].get(default="6")  # Selector doesnt have default field

`sel.xpath(...).get(default=...)` works in normal parsel without your changes.

Reply:
> About adding the strip argument to the other overloads, wouldn't that remove the purpose of the overload on those?

No, the purpose of those overloads is to show that the type of the return value depends on whether `default` is None or not.

> Wouldn't having an overloaded default + strip be just going back to the non-overloaded method signature?

I believe just adding `strip` to the existing overloads is enough, no new overloads are needed.

Reply:
`sel.xpath(...).get(default=...)` uses `SelectorList`, not `Selector`. The problem is that by indexing `[0]` then you go to a `Selector`, and it does not have `default` .
but either way I still have to give it a look

Reply:
Wait, you are right. `Selector.get()` is much simpler than `SelectorList.get()`. Well, I think it makes sense and is easy to add `strip=` to `Selector.get()`.

> it seems a little unintuitive to be using the [0] when doing a get operation, as you are already asking the SelectorList to get the first instance it found

Yeah, it's probably unusual to write `<SelectorList>[0].get()`. It's OK to have a Selector and call a get) on it though.

> I believe you should be able to do both default and strip directly on a Selector, but you can't for a Selector at the moment so maybe this needed further discussion.

Not so sure about default=, but maybe there are use cases where it's useful.

Reply:
I think we can keep this as is, strip for Selector.get() is less useful and can be implemented separately.